lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
@printercu `Why do you think it should not send mail when updating user right after create?` It's an unexpected behavior and in my opinion, it's a bug and we should fix it.